spec/system/consumer/split_checkout_details_spec.rb

```diff
@@ -0,0 +1,403 @@
+  let(:product) {
+    create(:taxed_product, supplier: supplier, price: 10, zone: zone, tax_rate_amount: 0.1)
+  }
+  let(:variant) { product.variants.first }
+  let!(:order_cycle) {
+    create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor],
+                                coordinator: create(:distributor_enterprise), variants: [variant])
+  }
+  let(:order) {
+    create(:order, order_cycle: order_cycle, distributor: distributor, bill_address_id: nil,
+                   ship_address_id: nil, state: "cart",
+                   line_items: [create(:line_item, variant: variant)])
+  }
+
+  let(:fee_tax_rate) { create(:tax_rate, amount: 0.10, zone: zone, included_in_price: true) }
+  let(:fee_tax_category) { create(:tax_category, tax_rates: [fee_tax_rate]) }
+  let(:enterprise_fee) { create(:enterprise_fee, amount: 1.23, tax_category: fee_tax_category) }
+
+  let(:free_shipping_with_required_address) {
+    create(:shipping_method, require_ship_address: true,
+                             name: "A Free Shipping with required address")
+  }
+  let(:free_shipping) {
+    create(:shipping_method, require_ship_address: false, name: "free Shipping",
+                             description: "yellow",
+                             calculator: Calculator::FlatRate.new(preferred_amount: 0.00))
+  }
+  let(:shipping_tax_rate) { create(:tax_rate, amount: 0.25, zone: zone, included_in_price: true) }
+  let(:shipping_tax_category) { create(:tax_category, tax_rates: [shipping_tax_rate]) }
+  let(:shipping_with_fee) {
+    create(:shipping_method, require_ship_address: true, tax_category: shipping_tax_category,
+                             name: "Shipping with Fee", description: "blue",
+                             calculator: Calculator::FlatRate.new(preferred_amount: 4.56))
+  }
+  let(:free_shipping_without_required_address) {
+    create(:shipping_method, require_ship_address: false,
+                             name: "Z Free Shipping without required address")
+  }
+  let(:tagged_shipping) {
+    create(:shipping_method, require_ship_address: false, name: "Local", tag_list: "local")
+  }
+  let!(:payment_with_fee) {
+    create(:payment_method, distributors: [distributor],
+                            name: "Payment with Fee", description: "Payment with fee",
+                            calculator: Calculator::FlatRate.new(preferred_amount: 1.23))
+  }
+  let(:shipping_backoffice_only) {
+    create(:shipping_method, require_ship_address: true, name: "Shipping Backoffice Only",
+                             display_on: "back_end")
+  }
+  let(:shipping_methods) {
+    [free_shipping_with_required_address, free_shipping, shipping_with_fee,
+     free_shipping_without_required_address, tagged_shipping]
+  }
+
+  before do
+    add_enterprise_fee enterprise_fee
+    set_order order
+
+    distributor.shipping_methods.push(shipping_methods)
+  end
+
```

Comment:
Question to reviewers: the lines within this setup block is repeated throughout the three new spec files. So I was wondering how to extract it to a helper file.

However, if one creates a such a helper file and then calls it on each of the specs, the variables are not being kept as they are declared with `let`. Is it worth having instance variables instead or is there a better approach? Or do we keep this repetition (does not sound right though...)

What do you think ?

Reply:
That's a good question. 
I think that in general it's worth having the setup in each file. It's likely that the setup will need to vary for each file, and it's much easier to refer to it at the top of the file. In fact, we might be able to cut down the setup to be more specific and efficient for each file (I think that a lot of the shipping records aren't used in all specs. For example `shipping_backoffice_only` is only used in`split_checkout_details_spec.rb`.).

However the current setup gives us a good, familiar base for all of the split_checkout_* specs. Keeping them consistent by moving to another file might make it easier to maintain them all in the long term. I think we could probably find a clean way to do that.

But I'm leaning towards just keeping them seperate like you have here. If we clean up the shipping `let`s, then there's not that much left and it's all pretty standard.

Reply:
Thanks for that @dacook , indeed the setup is a bit specific, so I went on and removed some bits which it seems we don't need (green build).

Also, I've renamed the files considering the future removal of the `split_` prefix, as per #11480 .

Do we need to update the (I believe untracked) [rubocop todo file](https://github.com/openfoodfoundation/openfoodnetwork/blob/4e05047cc669437c30f668887486d4c32e45cd57/.rubocop_todo.yml#L929)?

Reply:
> Do we need to update the (I believe untracked) [rubocop todo file](https://github.com/openfoodfoundation/openfoodnetwork/blob/4e05047cc669437c30f668887486d4c32e45cd57/.rubocop_todo.yml#L929)?

That would be ideal. I don't see it as a blocker for the PR but it would be nice.

Reply:
Sure gladly, sorry if it sounded like sort of an obvious question :-)

But I see that you've updated the file already. What's a bit confusing for me now is whether I should commit anything else: reason I ask is, if I run the script, then I get a different file then the one you've committed... Not too sure why this happens.